modules/identity/src/Volo.Abp.Identity.EntityFrameworkCore/Volo/Abp/Identity/EntityFrameworkCore/EfCoreOrganizationUnitRepository.cs

```diff
@@ -156,5 +156,37 @@ public override IQueryable<OrganizationUnit> WithDetails()
         {
             return GetQueryable().IncludeDetails();
         }
+
+        public virtual Task RemoveAllRolesAsync(
+            OrganizationUnit organizationUnit,
+            CancellationToken cancellationToken = default)
+        {
+            var ouRolesQuery = DbContext.Set<OrganizationUnitRole>()
+                .Where(q => q.OrganizationUnitId == organizationUnit.Id);
+
+            DbContext.Set<OrganizationUnitRole>().RemoveRange(ouRolesQuery);
+            return Task.FromResult(0);
+            //Can be long running process that could be made available for cancellation perhaps
+            //return Task.Run(() =>
+            //        DbContext.Set<OrganizationUnitRole>().RemoveRange(ouRolesQuery),
+            //        GetCancellationToken(cancellationToken)
+            //);
+        }
+
+        public virtual Task RemoveAllMembersAsync(
```

Comment:
must truly implement async. For example, you can use ToListAsync after Where instead of getting enumrable (I don't think RemoveRange handles it without performing the query).